index.js

```diff
@@ -63,8 +63,18 @@ function SentryLogger(options) {
     this.captureMessage = this.ravenClient.captureMessage
         .bind(this.ravenClient);
     this.sentryProber = options.sentryProber || null;
+    this.sentryProberCallbackImmediately = !options.sentryProberDetectFailuresBy;
```

Comment:
If the settings used to create the sentryProber are not supplied, log() will still callback immediately, otherwise it will wait for the result from the sentryProber via the specified method.